hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/ServiceFactories.java

```diff
@@ -89,9 +90,41 @@ private ServiceFactories() { }
      * @since 3.0
      */
     @Nonnull
-    public static <K, V> ServiceFactory<IMap<K, V>> iMapService(@Nonnull String mapName) {
-        return ServiceFactory
-                .withCreateFn(ctx -> ctx.jetInstance().<K, V>getMap(mapName))
-                .withLocalSharing();
+    public static <K, V> ServiceFactory<?, IMap<K, V>> iMapService(@Nonnull String mapName) {
+        return ServiceFactory.withCreateContainerFn(ctx -> ctx.jetInstance().<K, V>getMap(mapName))
+                             .withCreateServiceFn((ctx, map) -> map);
+    }
+
+    /**
+     * TODO
+     * @param createServiceFn
+     * @param destroyServiceFn
+     * @param <S>
+     * @return
+     */
+    public static <S> ServiceFactory<?, S> perInstanceService(
+            @Nonnull SupplierEx<S> createServiceFn,
```

Comment:
Why don't we give access to the `ProcessorSupplier.Context` here and to `Processor.Context` in `perProcessorService`?